core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java

```diff
@@ -186,18 +202,38 @@ public void completeFile(CompleteFilePRequest request,
   public void createDirectory(CreateDirectoryPRequest request,
       StreamObserver<CreateDirectoryPResponse> responseObserver) {
     CreateDirectoryPOptions options = request.getOptions();
+
     RpcUtils.call(LOG, () -> {
+      if (request.getOptions().getCommonOptions().getCheckS3BucketPath()) {
+        checkBucketPathExists(request.getPath());
+      }
       AlluxioURI pathUri = getAlluxioURI(request.getPath());
       mFileSystemMaster.createDirectory(pathUri, CreateDirectoryContext.create(options.toBuilder())
           .withTracker(new GrpcCallTracker(responseObserver)));
       return CreateDirectoryPResponse.newBuilder().build();
     }, "CreateDirectory", "request=%s", responseObserver, request);
   }
 
+  private void checkBucketPathExists(String path)
+      throws AlluxioException, IOException {
+
+    String bucketPath =
+        AlluxioURI.SEPARATOR + path.split(AlluxioURI.SEPARATOR, 3)[1];
+    boolean exists = mFileSystemMaster.exists(getAlluxioURI(bucketPath),
+        ExistsContext.create(ExistsPOptions.getDefaultInstance().toBuilder()));
+    if (!exists) {
+      throw new FileDoesNotExistException("Bucket " + bucketPath
```

Comment:
In lines 220~221, would it be possible to throw exceptions lie Array Index Out Of Bounds? In that case, should it also be considered as FileDoesNotExistException? Or we just simply throw to caller out of bound? I don't have strong opinion, just want to bring that for you to think about it.